### DISCUSSION TITLE
node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node) whilst trying to deploy

### DISCUSSION THREAD

**Discussion by tobyab** [vercel/community#2867] [Help] [answered]
### Summary

I'm running into this error whilst trying to deploy to Vercel. I'm totally stuck and I'd love some help!

```
node: /lib64/libm.so.6: version `GLIBC_2.27' not found (required by node)
node: /lib64/libc.so.6: version `GLIBC_2.28' not found (required by node)
Error: Command "yarn prisma generate && next build" exited with 1
BUILD_UTILS_SPAWN_1: Command "yarn prisma generate && next build" exited with 1
```

### Steps to Reproduce

Deploy project to Vercel. My build command is `yarn prisma generate && next build`. Sorry! This isn't much help. 😅

**Comment by smaeda-ks** [comment]
@developedbytoby Are you trying to use Canvas or something like that? If that's the case, you may find some useful tips in this issue: https://github.com/vercel/vercel/issues/3460

there's also a good alternative solution: https://github.com/Brooooooklyn/canvas

**Reply by tobyab** [reply]
I'm not. Here's a list of all of my dependencies: 
```
 "@next-auth/prisma-adapter": "^1.0.7",
    "@paralleldrive/cuid2": "^2.2.1",
    "@prisma/client": "^4.15.0",
    "@sentry/nextjs": "^7.38.0",
    "@types/react-modal": "^3.13.1",
    "@vercel/analytics": "^0.1.10",
    "autoprefixer": "^10.4.13",
    "checkly": "^4.0.8",
    "classnames": "^2.3.2",
    "contentlayer": "^0.3.0",
    "copy-to-clipboard": "^3.3.3",
    "csstype": "^3.1.1",
    "date-fns": "^2.29.3",
    "eslint-plugin-react": "^7.32.2",
    "framer-motion": "^10.11.2",
    "mailing-core": "^1.0.0",
    "million": "1.12.6-beta.2",
    "next": "^13.2.4",
    "next-auth": "^4.22.1",
    "next-contentlayer": "^0.3.0",
    "node": "^19.6.1",
    "nodemailer": "^6.9.3",
    "postcss": "^8.4.21",
    "prisma": "^4.15.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-hot-toast": "^2.4.0",
    "react-modal": "^3.16.1",
    "react-wrap-balancer": "^0.4.0",
    "remark": "^14.0.3",
    "remark-html": "^15.0.2",
    "resend": "^0.11.0",
    "stripe": "^12.3.0",
    "swr": "^2.1.5",
    "tailwindcss": "^3.3.2",
```

**Reply by smaeda-ks** [reply, marked as answer]
@developedbytoby so the issue is you're including `node` itself in your `dependencies`... that's the problem. You should not include that.

**Reply by smaeda-ks** [reply]
This is how you define your Node version for your Vercel projects:
https://vercel.com/docs/concepts/functions/serverless-functions/runtimes/node-js#node.js-version

**Reply by tobyab** [reply]
Oh shoot! Completely missed that. The node versions are mismatched. Thank you!!!

**Reply by davidkartuzinski** [reply]
For anyone getting this error, the error disappears when you delete the node version as stated above. I had to reread it and check other solutions first before I realized its really as simple as deleting the node version line from package.json.

**Reply by marsidorowicz** [reply]
thanks for help!

**Reply by lancejpollard** [reply]
@smaeda-ks I am using Node v20.10.0 locally, but Node 18 on Vercel, and getting this error still. I don't have `node` defined in my package.json anywhere. Could it be that because I am using Node 20 locally that when I `pnpm add react-pdf` (which uses node-canvas), it installs some different version of the package because I am using Node.js 20, so then when I deploy to Vercel with Node 18, there is a mismatch?

In that case I would have to figure out how to install Node 18 locally and reinstall my packages, but not sure that I have the thought process right, not sure that it matters. Any ideas?

BTW I was able to install on Node 20 on Vercel just fine, no hacks. So not sure. But I am getting other weird behavior with Node 20 so far on Vercel (like it hangs when `yum install x` anything, but beside the point). So can't really use Node 20 atm either.

**Reply by brunogabado** [reply]
Thanks mate !

**Reply by sauravhathi** [reply]
Issue fixed! ✅

In my case, I am using **KonvaJS** where I encountered the same issue.

I simply installed `'node' `and it works for me.

**Reply by vishal590** [reply]
I removed node but still got same error

**Reply by vishal590** [reply]
yeah, it worked
I removed node from package.json

**Reply by saijoro** [reply]
can i include the types/node in dev-dependencies for typescript usage in package lock json

**Comment by steddyman** [comment]
I ahve the same problem and my node versions are not mismatched.  I do not have "node" defined as a dev dependency.  My Node JS versions in the portal and in the project are both 18.6.1, but I still get the same GLIBC error.

**Reply by teppakornt** [reply]
I change the version node on Vercel project setting to 16.X and everything works fine.
![image](https://github.com/vercel/community/assets/26103242/d43b2a69-0678-46b0-b658-55c0cc7b49f9)

**Reply by smaeda-ks** [reply]
https://github.com/orgs/vercel/discussions/3061#discussioncomment-6315667

**Reply by Klastic** [reply]
Downgrading to Node.js 16.x as suggested didn't work for me due to compatibility issues with Next.js 14. Instead, upgrading to Node.js 20.x resolved the issue.
